pkg/chunk/disk_cache.go

```diff
@@ -615,6 +615,48 @@ func (cache *cacheStore) load(key string) (ReadCloser, error) {
 	return f, err
 }
 
+func (cache *cacheStore) exist(key string) (bool, error) {
+	cache.Lock()
+	defer cache.Unlock()
+	if _, ok := cache.pages[key]; ok {
+		return true, nil
+	}
+	k := cache.getCacheKey(key)
+	if cache.scanned && cache.keys[k].atime == 0 {
+		return false, errNotCached
+	}
+	cache.Unlock()
+	var fi os.FileInfo
+	var err error
+	err = cache.checkErr(func() error {
+		fi, err = os.Stat(cache.cachePath(key))
+		return err
+	})
+
+	cache.Lock()
+	if err == nil {
+		if it, ok := cache.keys[k]; ok {
+			// update atime
+			nowTime := time.Now()
+			err = cache.checkErr(func() error {
+				err = os.Chtimes(cache.cachePath(key), nowTime, fi.ModTime())
```

Comment:
After this, this one could be slower than before if the disk is mounted with noatime.

I think we can ignore this, the atime in memory could be enough.